### PR TITLE
`Communication`: Fix an issue with duplicated posts on course wide search

### PR DESCRIPTION
**Pull request by cremertim**
#### General

- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [ ] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Server
- [ ] **Important**: I implemented the changes with a [very good performance](https://docs.artemis.cit.tum.de/dev/guidelines/performance/) and prevented too many (unnecessary) and too complex database calls.
- [ ] I **strictly** followed the principle of **data economy** for all database calls.
- [ ] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [ ] I documented the Java code using JavaDoc style.

### Motivation and Context

Currently, the search returns for each answer post an occurences in the course wide search, since we do not filter on duplicates. This PR fixes the issue

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to the Communication Section of a Course
3. Search for an exisiting post that has answer posts
4. Verify, the exisiting post is only shown once in the search

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Performance Review

- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance even for very large courses with more than 2000 students.
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance even for very large courses with more than 2000 students.
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test 1
- [ ] Test 2
#### Performance Tests
- [ ] Test 1
- [ ] Test 2

### Screenshots

Before:
image

After:
image

## Summary by CodeRabbit

- **New Features**
	- Enhanced post retrieval to ensure only unique post IDs are returned, improving data accuracy and reducing redundancy.

### CODE REVIEW

**Review by sachmii — APPROVED** [review]
Tested on TS5, worked just fine. No duplicated were shown

**Review by HawKhiem — APPROVED** [review]
Tested on TS5. Works as described. Post only show up once

File: src/main/java/de/tum/cit/aet/artemis/communication/repository/CustomPostRepositoryImpl.java [hunk 1]
```diff
@@ -50,6 +52,8 @@ public Page<Long> findPostIdsWithSpecification(Specification<Post> specification
         query.setMaxResults(pageable.getPageSize());
 
         List<Long> postIds = query.getResultList();
+        // removes all duplicates from the answer posts
+        List<Long> uniquePostIds = new ArrayList<>(new LinkedHashSet<>(postIds));
```
Comment (MaximilianAnzinger) [inline src/main/java/de/tum/cit/aet/artemis/communication/repository/CustomPostRepositoryImpl.java:56]: Why do we even retrieve a list and not directly a set here?

**Review by MaximilianAnzinger — CHANGES_REQUESTED** [review]

**Review by vinceclifford — APPROVED** [review]
Tested on TS5, works as expected.

**Review by Cathy0123456789 — APPROVED** [review]
Tested on TS5. Works as described

File: src/main/java/de/tum/cit/aet/artemis/communication/repository/CustomPostRepositoryImpl.java (hunk 1 above)
Comment (cremertim) [reply src/main/java/de/tum/cit/aet/artemis/communication/repository/CustomPostRepositoryImpl.java:56]: I now tried to return the set as a list by using distinct, but this is not possible. The search specification always contains a order by clause which leads to an hibernate error (by not returning the order-property).

**Review by ItsaaaMeMario — APPROVED** [review]
Tested on TS5, search result only shows up once, works as expected.

**Review by zagemello — APPROVED** [review]
Tested on TS5, all works as expected.

**Review by MaximilianAnzinger — APPROVED** [review]
Thx for looking into it again 👍

**Review by krusche — APPROVED** [review]

**Pull request merged**